### ISSUE TITLE
Metal samples crashes with: sg_apply_draw_state: pipeline object required

### ISSUE THREAD

**Issue by Oldes** [floooh/sokol-samples#3]
The samples for Metal config on OSX (v10.12.6) seems not to be working. This is what I get when I try to run them:
```
Davids-MBP:sokol-samples oldes$ ./fips run triangle-metal
=== run 'triangle-metal' (config: metal-osx-xcode-debug, project: sokol-samples):
2017-12-11 23:01:44.905 triangle-metal[45198:2300054] -[MTLRenderPipelineDescriptorInternal vertexBuffers]: unrecognized selector sent to instance 0x7fb8fed16f90
2017-12-11 23:01:44.905 triangle-metal[45198:2300054] -[MTLRenderPipelineDescriptorInternal vertexBuffers]: unrecognized selector sent to instance 0x7fb8fed16f90
sg_apply_draw_state: pipeline object required
^^^^  VALIDATION FAILED, TERMINATING ^^^^
Assertion failed: (0), function _sg_validate_end, file /Users/oldes/GIT/sokol/_sokol_gfx.impl.h, line 830.
```

**Comment by floooh** [comment]
Thanks for the report, not sure yet what would cause this. I'm on MacOS 10.13.2 with Xcode 9.2 on a mid-2014 13" MBP (with Intel GPU).

The error seems to be that an unknown Objective-C method is called on some Metal object when setting vertex buffers. May be I'm accidently calling a method that's not available OSX 10.12 or an older SDK version.

I'm not planning to support the most recent OSX/Metal versions only, so I guess I should fix this :)

**Comment by floooh** [comment]
Ah ok, here's the problem: https://developer.apple.com/documentation/metal/mtlrenderpipelinedescriptor/2879280-vertexbuffers?language=objc

The method I'm calling is only available since OSX 10.13. Good catch :) I'll try to fix this ASAP.

**Comment by floooh** [comment]
I added a quick fix in sokol (https://github.com/floooh/sokol/commit/c4623b95d9a1ce53558b93fe9ec0a335b1fca93b). Hopefully this was the only case the prevents running on older macOS version, if you find anything else, please open a new ticket :)
